clang/lib/AST/CommentParser.cpp

```diff
@@ -89,6 +89,31 @@ class TextTokenRetokenizer {
     }
   }
 
+  /// Extract a template type
+  bool lexTemplate(SmallString<32> &WordText) {
+    unsigned IncrementCounter = 0;
+    while (!isEnd()) {
+      const char C = peek();
+      WordText.push_back(C);
+      consumeChar();
```

Comment:
It would be nice to modify `consumeChar` to return the consumed char

Reply:
I would prefer to do that as a separate refactor PR, I think it's outside the scope of this change.